### ISSUE TITLE
Bower support + Firefox NX source download bug

### ISSUE THREAD

**Issue by ghost** [RisingStack/nx-framework#13]
I need to clarify some questions:
1) Is it possible to install it using Bower package? or only NPM?
2) Why does it include GA module? 
3) And takes 300kb in size?
4) How soon IE11 support can be added?

**Comment by krnlde** [comment]
1. Yes, someone will have to register it, preferably @solkimicreb 
2. GA?
3. no it takes up 96kb
4. As soon as Shadow DOM and Proxies are implemented or polyfilled for IE11, so most likely never
5. what do these questions have to do with knockout whatsoever?

**Comment by ghost** [comment]
GA?
Yes GA!
https://www.nx-framework.com/downloads/nx-beta.2.0.0.js

    <script>window.ga = window.ga || ((...args) => (ga.q = ga.q || []).push(args))</script>

Spyware detected?     

nx-beta.2.0.0.js   
takes 288 KB    

what do these questions have to do with knockout whatsoever?   
all of that is present in KO, including IE6 support... but we need only IE11.

**Comment by NetOpWibby** [comment]
@hardhub GA is Google Analytics. Whether you think that's spyware because it's Google is up to you, lol.

**Comment by ghost** [comment]
@NetOperatorWibby 

LOL?

LOL or lol is acronym for laugh out loud... why are you laughing without a reason? Are you idiot?

It was not mentioned as real malware.. But behavior of the framework is the same as for spyware. This framework tracks users of all apps loading mentioned js-file with GA... But framework is NOT intended to do such things... It is UI framework.. Why should I inject framework GA in my app? It is even never mentioned in any documents... Privacy of users of my app are not important? Then I say this library goodbye. It is not the way for OSS. Even M$ claims about opt out telemetry in .net core tools... bad BUT NOT HIDDEN and with option to DISABLE. So yes it looks really like a Trojan horse!

**Comment by vivekimsit** [comment]
Guys it would be great if we all could behave with maturity and keep away from using foul languages. If GA seems a big issue then please raise an issue(which is already done here) and let the library maintainers look into it.

@solkimicreb can you please look into it?

**Comment by solkimicreb** [comment]
Hi @hardhub !

The framework is 12kb minzipped if everything is included (the next version will be minified by default). Unminified and unzipped it is 92kb. **It definitely has no GA inside it.** What you opened is the source code of the nx **home page** (yes, the home page has GA tracking). The link that you provided is a download link, if you open it in the browser it points to the source of the nx home page. If you open it by clicking the actual download link [here](https://www.nx-framework.com/install), it sets the correct headers and points to the nx source code. This is done to work around a problem in Safari. I hope it clarified things for you.

Bower is not supported yet. I will try to add it to the next release (scheduled in approx a week).
IE11 support won't happen in the immediate future. The core could easily support IE, but many of the middlewares can't be polyfilled for IE. 

Edit: If you are interested in the source code check the repos in [this GitHub organization](https://github.com/nx-js)

**Comment by ghost** [comment]
Hi @solkimicreb 

Now all much more clear. But I think it does not work as you described it.
When I click "NX beta 2.0.0 (recommended)" it downloads https://www.nx-framework.com/downloads/nx-beta.2.0.0.js which is 288 Kb in size.

At least it happens in firefox v51.

**Comment by solkimicreb** [comment]
You are right, it does happen in Firefox. I checked some other browsers and it works correctly in Chrome, Safari and Opera. I think that the download hack, that I used for Safari broke it. I will look into it tomorrow.

Thanks for the bug catch (:

**Comment by NetOpWibby** [comment]
@hardhub I wasn't laughing at you, it was more for me as *I* often view Google as malware. Like @vivekimsit mentioned, your language was unnecessary and your reaction was overboard.

**Comment by ghost** [comment]
@NetOperatorWibby 

>I wasn't laughing at you, it was more for me as I often view Google as malware.

All what is for you should be at you... I think reasonable...

> Like @vivekimsit mentioned, your language was unnecessary and your reaction was overboard.

It is not for you to decide... I do not like your reaction no less.

But I do not think we need to discuss it anymore... Originally I told about technical issues.. they are now clear.. so we just need to solve it... nothing else...

**Comment by solkimicreb** [comment]
I will keep this issue open until the firefox download bug is fixed and bower support is added. I changed the issue title a bit, @hardhub I hope you don't mind.

**Comment by solkimicreb** [comment]
Firefox issue is fixed, the next release will take some more time though (It got bigger then I originally planned). I will keep the issue open till bower support lands.
